### PR TITLE
Catch stress thread errors and generate critical event on em

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/5GAiNOHY/1573-catch-errors-happening-during-stress-command-run-by-creating-an-critical-event
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/stress_thread.py [hunk 1]
```diff
@@ -158,11 +158,23 @@ def _run_stress(self, node, loader_idx, cpu_idx, keyspace_idx):
...
-                                      ignore_status=True,
-                                      log_file=log_file_name)
+            result = None
+            try:
+                result = node.remoter.run(cmd=node_cmd,
+                                          timeout=self.timeout,
+                                          ignore_status=True,
+                                          log_file=log_file_name)
+            except Exception as exc:  # pylint: disable=broad-except
+                CassandraStressEvent(type='failed', node=str(node), stress_cmd=stress_cmd,
```
Comment (bentsi) [inline sdcm/stress_thread.py]: better `type='failure'`
since start, finish are nouns

Comment (fruch) [reply sdcm/stress_thread.py]: Won't it be enough just to change the `remoter.run()` to use `ignore_status=False` ?
And then we could only catch exceptions, and not have this duplication of events sending code

File: sdcm/stress_thread.py [hunk 2]
```diff
@@ -158,11 +158,23 @@ def _run_stress(self, node, loader_idx, cpu_idx, keyspace_idx):
...
+                result = node.remoter.run(cmd=node_cmd,
+                                          timeout=self.timeout,
+                                          log_file=log_file_name)
+            except Exception as exc:  # pylint: disable=broad-except
+                #  pylint: disable=no-member
+                if hasattr(exc, 'result') and exc.result.failed:
+                    output = exc.result.stderr
+                    if not output:
+                        output = exc.result.stdout
+                    errors_str = f'Stress command return error: {output}'
```
Comment (bentsi) [inline sdcm/stress_thread.py]: this is bad! we store the output of the command in the log and it is huge! why do you want to output again and in the event? Anyway, I don't understand why are you complicating this? this case can be finished by:

```
CassandraStressEvent(type='finish', node=str(node), stress_cmd=stress_cmd,
                                     log_file_name=log_file_name, severity=Severity.CRITICAL,
                                     errors=errors_str, severity=Severity.CRITICAL if result.exit_status else  Severity.NORMAL)
```
(the type is not so important, currently, we don't use it)

**Comment by dkropachev** [comment]
@bentsi , fixed

File: sdcm/stress_thread.py [hunk 3]
```diff
@@ -213,10 +222,18 @@ def get_results(self):
...
+                lines = output.splitlines()
+                node_cs_res = BaseLoaderSet._parse_cs_summary(lines)  # pylint: disable=protected-access
+                if node_cs_res:
+                    ret.append(node_cs_res)
+            except Exception as exc:  # pylint: disable=broad-except
+                if lines:
+                    lines = '\n'.join(lines)
+                CassandraStressEvent(
+                    type='failure', node='', severity=Severity.CRITICAL,
+                    errors=f'Failed to proccess stress command output due to {exc} error\n data: {lines}')
```
Comment (bentsi) [inline sdcm/stress_thread.py]: `lines` can be too big to print, please remove it

**Comment by bentsi** [comment]
Dmitry, this simple PR is taking too long. When fixing things don't revert previous fixes and **please review the code before you push**!

Maybe I wasn't clear in my comment,  you can't attach the stdout of cassandra-strass process in the event string, it is simply **too big** and not needed in the event.

**Comment by bentsi** [comment]
can't backport tot 3.2, 2019, since they use Python 2

**Pull request merged**